### DISCUSSION TITLE
Quarkus RestEasy Reactive Not Exploding List Type Query Parameter

### DISCUSSION THREAD

**Discussion by The-Funk** [quarkusio/quarkus#27115] [Q&A] [answered]
I have a query parameter declared as follows in my method (@RestQuery String[] someList)

However when I go to reference this data, the array is converted to an array of 1 String.

Am I doing something wrong here? Is it because of how I'm entering my query parameters? I tend to use the shorthand comma separated value notation and only ref the parameter name one time in the URL. I prefer this in order to keep the URL shorter. For instance www.example.com?someList=value1,value2,value3 as opposed to www.example.com?someList=value1&someList=value2&someList=value3

[a comment by an automated account is not shown]

**Comment by geoand** [comment]
Have you tried using a `List` instead of an Array: `@RestQuery List<String> someList` ?

**Reply by The-Funk** [reply]
Just tried today. Same result for List<String> if query parameters are provided in short format as shown above.

**Reply by gsmet** [reply]
Is it supposed to work and automatically explode the value on commas?

**Reply by geoand** [reply]
No, it's not. `List` is for capturing multiple values (for example when doing something like `?k1=v1&k1=v11`

**Reply by The-Funk** [reply]
Both List<String> and String[] work when supplying multiple values via the ```?k1=v1&k1=v2``` method of supplying a query parameter with multiple values, however neither explode comma separated values when provided like this ```?k1=v1,v2```.

**Reply by geoand** [reply]
That's how JAX-RS works unfortunately

**Reply by The-Funk** [reply]
Is there no support for the shorthand notation at all? How do people provide comma separated lists in their URLs? The shorthand is much more efficient, I try to use it whenever possible. Right now I'm just manually exploding the results.

**Reply by geoand** [reply]
Manual is the only way

**Reply by The-Funk** [reply]
:(

**Comment by MichaelCkr** [comment, marked as answer]
Just for documentation, if someone else stumbles over this problem and request here:

Here is the discussion https://github.com/quarkusio/quarkus/discussions/29500 leading to this really nice improvement https://github.com/quarkusio/quarkus/issues/29528 enabling

```java
@GET
public String hello(@QueryParam("foo") @Separator(",") List<String> foo) {}
```

**Reply by The-Funk** [reply]
Well that's cool! 😃
